content/en/blog/2024/inpod-traffic-redirection-ambient/index.md

```diff
@@ -0,0 +1,229 @@
+
+The [Istio-cni](/docs/setup/additional-setup/cni/) component is an optional component in the sidecar data plane mode,
+commonly used to remove the [requirement for the `NET_ADMIN` and `NET_RAW` capabilities](/docs/ops/deployment/requirements/) for
+users deploying pods into the Istio mesh. With the launch of ambient, istio-cni has become a required component for ambient
+data plane mode. Whenever pods are added to ambient mesh, the istio-cni component configures traffic redirection for all
+incoming and outgoing traffic between the application pods and their co-located [ztunnel](/blog/2023/rust-based-ztunnel/) on
+the node via the node-level network namespace. The key difference between sidecar and ambient on the traffic redirection is
+the redirection in ambient is across network namespaces, from the application namespace to the Istio’s system namespace
+where ztunnel typically runs.
+
+As we tested more broadly in multiple Kubernetes environments which many of them have its default CNI, it became clear that
+the host network based traffic redirection mechanism we had been using for Istio ambient traffic capture and redirection
+during the alpha stage was not going to meet our requirements. The following challenges were identified:
+- The default CNI’s node-level networking configuration could interfere with the node-level networking configuration
+from istio-cni, whether it is with eBPF or iptables.
+- If users deploy a network policy for the default CNI, the network policy may not be enforced when istio-cni is deployed.
+
+### Thought process on solving the challenges
+
+At Solo, we have been brainstorming a lot on these top challenges in Ambient. We realized that applying any traffic
+routing/networking rules in the node-level network namespace invites unresolvable conflicts/incompatibilities with
+CNIs (which might use a wide variety of fundamentally networking topologies on the node) - they *must* configure traffic
+routing/networking rules in the node-level network namespace. We realized any eBPF implementation would have the same
+basic problem, as there is no standardized way to safely chain/extend arbitrary eBPF programs at this time.
+
+With sidecar data plane mode, it is intuitive to configure traffic redirection between sidecar and application pod within
+the pod network namespace. One day, a light-bulb moment arrived - why not mimic sidecars and configure the redirection in
+the application network space? While this sounds a crazy simple thought, is this technically possible when ztunnel runs
+in the Istio’s system namespace? After a lot of research, we realized a Linux process running in one network namespace
+could create and own listening sockets within another network namespace, which is a basic Linux socket capability.
+However, to make this work and cover all pod lifecycle scenarios we had to make architectural changes to the ztunnel
+as well as to the istio-cni agent.
+
+### Traffic redirection in ambient now
+
```

Comment:
I don't know if we need another heading?

Reply:
Agree, I don't see a lot of benefit personally.

Reply:
removed